src/utils/space.tsx

```diff
@@ -71,15 +71,15 @@ export const showAddExistingRooms = (space: Room): void => {
     });
 };
 
-export const showCreateNewRoom = async (space: Room, type?: RoomType): Promise<boolean> => {
+export const showCreateNewRoom = async (space: Room, type?: RoomType, favourite?: boolean): Promise<boolean> => {
     const modal = Modal.createDialog<[boolean, IOpts]>(CreateRoomDialog, {
         type,
-        defaultPublic: space.getJoinRule() === JoinRule.Public,
+        defaultPublic: space? space.getJoinRule() === JoinRule.Public : undefined,
```

Comment:
space is not optional in the type signature, so why this check? If space can be falsy, the signature should change to reflect that.

Reply:
Also, `CreateRoomDialog` doesn't differentiate between falsy and strictly false, so why pass undefined? If `space` is made an optional param then using `defaultPublic: space?.getJoinRule() === JoinRule.Public` works the same.

Reply:
> Also, `CreateRoomDialog` doesn't differentiate between falsy and strictly false, so why pass undefined? If `space` is made an optional param then using `defaultPublic: space?.getJoinRule() === JoinRule.Public` works the same.

Because if `CreateRoomDialog` ever decided to default to `true` for `defaultPublic`, then `undefined` would be more correct